wordpress.org/public_html/wp-content/plugins/wporg-gp-customizations/inc/class-plugin.php

```diff
@@ -268,12 +287,62 @@ public function auto_reject_already_rejected( $args, GP_Translation $translation
 					}
 				}
 			}
-
 		}
 
 		return $args;
 	}
 
+	/**
+	 * Stores source of a translation in database.
+	 *
+	 * @param GP_Translation $translation Translation instance.
+	 * @return void
+	 */
+	public function log_translation_source( GP_Translation $translation ) {
+		$source = '';
+		if ( 'GP_Route_Translation' === GP::$current_route->class_name ) {
+			if ( 'import_translations_post' === GP::$current_route->last_method_called ) {
+				$http_referer                     = stripslashes( $_POST['_wp_http_referer'] );
+				$this->imported_translation_ids[] = $translation->id;
+				$this->imported_source            = preg_match( '/^\.\.\/\?sort/', $http_referer ) ? 'import_from_playground' : 'import_from_frontend';
+				return;
+			}
+			if ( 'translations_post' === GP::$current_route->last_method_called ) {
+				if ( isset( $_POST['translation_source'] ) ) {
+					$source = sanitize_text_field( $_POST['translation_source'] );
+				}
+			}
+		}
+		if ( ! $source ) {
+			return;
+		}
+		gp_update_meta( 0, $translation->id, $source, 'gp_option' );
```

Comment:
/**
 * Adds and updates meta data in the database
``` *
 * @since 1.0.0
 *
 * @param int    $object_id  ID of the object metadata is for.
 * @param string $meta_key   Metadata key.
 * @param mixed  $meta_value The value to store.
 * @param string $type       The object type.
 * @param bool   $global     Overrides the requirement of $object_id to be a number OR not empty.
 *
 * @return bool|int True if meta updated, false if there is an error and the id of the inserted row otherwise.
 */
function gp_update_meta( $object_id = 0, $meta_key, $meta_value, $type, $global = false ) {
```

```suggestion
		gp_update_meta( $translation->id, 'source', $source, 'translation' );
```